YARN-10928. Support default queue config for minimum-user-limit-percent/user-limit-factor

zheng-weihao opened the pull request:
<!--
  Thanks for sending a pull request!
    1. If this is your first time, please read our contributor guidelines: https://cwiki.apache.org/confluence/display/HADOOP/How+To+Contribute
    2. Make sure your PR title starts with JIRA issue id, e.g., 'HADOOP-17799. Your PR title ...'.
-->

### Description of PR
There are many user cases that one user owns many queues in his organization's cluster for different business usages in practice. These queues often share the same properties, such as minimum-user-limit-percent and user-limit-factor. Users have to write one property for every queue they use if they want to use customized these shared properties. Adding default queue properties for these cases will simplify capacity scheduler's configuration file and make it easy to adjust queue's common properties. 
 
CHANGES:
Add two properties as queue's default value in capacity scheduler's configuration:
`yarn.scheduler.capacity.minimum-user-limit-percent`
`yarn.scheduler.capacity.user-limit-factor`

### How was this patch tested?


### For code changes:

- [x] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

Commit:
Support default queue config for minimum-user-limit-percent/user-limit-factor

hadoop-yetus commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 52s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
|||| _ trunk Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  32m  9s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   1m  8s |  |  trunk passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  compile  |   0m 57s |  |  trunk passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  checkstyle  |   0m 51s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   1m  4s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   0m 50s |  |  trunk passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javadoc  |   0m 44s |  |  trunk passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  spotbugs  |   2m  0s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  15m 57s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 55s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  |  the patch passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javac  |   1m  0s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 49s |  |  the patch passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  javac  |   0m 49s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 41s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   0m 53s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 39s |  |  the patch passed with JDK Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04  |
| +1 :green_heart: |  javadoc  |   0m 36s |  |  the patch passed with JDK Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10  |
| +1 :green_heart: |  spotbugs  |   2m  4s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  15m 42s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  96m 55s |  |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 33s |  |  The patch does not generate ASF License warnings.  |
|  |   | 176m 39s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3356/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3356 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 01ab223f8b6c 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 365453676718f62e21ca5040779483d08e5a22e6 |
| Default Java | Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.11+9-Ubuntu-0ubuntu2.20.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_292-8u292-b10-0ubuntu1~20.04-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3356/1/testReport/ |
| Max. process+thread count | 989 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3356/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

dongjoon-hyun committed:
HADOOP-17869. `fs.s3a.connection.maximum` should be bigger than `fs.s3a.threads.max` (#3337).


The value of `fs.s3a.connection.maximum` has been increased to 96

Contributed by Dongjoon Hyun

umamaheswararao committed:
HDFS-16192: ViewDistributedFileSystem#rename wrongly using src in the place of dst. (#3353)

Co-authored-by: Uma Maheswara Rao G <umagangumalla@cloudera.com>

yellowflash committed:
HADOOP-17870. Http Filesystem to qualify relative paths. (#3338)


Contributed by Yellowflash

lzx404243 committed:
MAPREDUCE-7311. Clear filesystem statistics after tests in TestTaskProgressReporter (#2500)

Co-authored-by: Zhengxi Li <zli89@illinois.edu>

langlaile1221 committed:
HDFS-15966. Empty the statistical parameters when emptying the redundant queue (#2894)

amahussein committed:
HADOOP-17886. Upgrade ant to 1.10.11 (#3371)

virajjasani committed:
HADOOP-17874. ExceptionsHandler to add terse/suppressed Exceptions in thread-safe manner (#3343)

Signed-off-by: Akira Ajisaka <aajisaka@apache.org>

WeisonWei committed:
HDFS-16202. Use constants "HdfsClientConfigKeys.Failover.PREFIX" instead of "dfs.client.failover." (#3367). Contributed by Weison Wei.

Signed-off-by: Ayush Saxena <ayushsaxena@apache.org>

cnauroth committed:
HADOOP-15129. Datanode caches namenode DNS lookup failure and cannot startup (#3348)
Co-authored-by:  Karthik Palaniappan

Change-Id: Id079a5319e5e83939d5dcce5fb9ebe3715ee864f

virajjasani committed:
HDFS-16199. Resolve log placeholders in NamenodeBeanMetrics (#3362). Contributed by Viraj Jasani.

Signed-off-by: Ayush Saxena <ayushsaxena@apache.org>

tomscut committed:
HDFS-16194. Simplify the code with DatanodeID#getXferAddrWithHostname (#3354). Contributed by  tomscut.

prasad-acit committed:
HDFS-16138. BlockReportProcessingThread exit doesn't print the actual stack (#3244). Contributed by  Renukaprasad C.

Commit:
Merge branch 'YARN-10928' of github.com:FrankinRUC/hadoop into YARN-10928

Commit:
Add a unit test and document details for these properties
